x/rollapp/keeper/invariants.go

```diff
@@ -252,3 +257,52 @@ func RollappFinalizedStateInvariant(k Keeper) sdk.Invariant {
 		), broken
 	}
 }
+
+// LivenessEventInvariant checks for all rollapps that the liveness event height, if any, is accurate,
+// in that there is actually an event stored at that height. Moreover, there should not be any events
+// stored which don't correspond to a liveness event height.
+func LivenessEventInvariant(k Keeper) sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+		var (
+			broken bool
+			msg    string
+		)
+		rollapps := k.GetAllRollapps(ctx)
+		for _, ra := range rollapps {
+			if !ra.LastStateUpdateHeightIsSet() {
+				continue
+			}
+			events := k.GetLivenessEvents(ctx, &ra.LivenessEventHeight)
+			cnt := 0
+			for _, event := range events {
+				if event.RollappId == ra.RollappId {
+					cnt++
+				}
+			}
+			if cnt != 1 {
+				broken = true
+				msg += fmt.Sprintf("| rollapp stored event but wrong number found in queue: rollapp: %s: event height: %d: found: %d", ra.RollappId, ra.LivenessEventHeight, cnt)
```

Comment:
Not critical, but it would have been more appropriate to use `strings.Builder` here. wdyt?

Reply:
yeah
could say same for all the invariants